klayout/drc/rule_decks/main.drc

```diff
@@ -14,880 +14,763 @@
+  if $topcell
+    source($input, $topcell)
+  else
+    source($input)
+  end
 end
 
-
-logger.info("Loading database to memory is complete.")
+logger.info('Loading database to memory is complete.')
 
 if $report
-    logger.info("GF180MCU Klayout DRC runset output at: %s" % [$report])
-    report("DRC Run Report at", $report)
+  logger.info("GF180MCU Klayout DRC runset output at: #{$report}")
+  report('DRC Run Report at', $report)
 else
-    layout_dir = Pathname.new(RBA::CellView::active.filename).parent.realpath
-    report_path = layout_dir.join("gf180_drc.lyrdb").to_s
-    logger.info("GF180MCU Klayout DRC runset output at default location: %s" % report_path)
-    report("DRC Run Report at", report_path)
+  layout_dir = Pathname.new(RBA::CellView.active.filename).parent.realpath
+  report_path = layout_dir.join('gf180_drc.lyrdb').to_s
+  logger.info('GF180MCU Klayout DRC runset output at default location: %s', report_path)
+  report('DRC Run Report at', report_path)
 end
 
 #================================================
 #------------------ SWITCHES --------------------
 #================================================
-logger.info("Evaluate switches.")
+logger.info('Evaluate switches.')
 
 # connectivity rules
-if $conn_drc == "true"
+if $conn_drc == 'true'
   CONNECTIVITY_RULES = $conn_drc
-  logger.info("connectivity rules are enabled.")
+  logger.info('connectivity rules are enabled.')
 else
   CONNECTIVITY_RULES = false
-  logger.info("connectivity rules are disabled.")
-end # connectivity rules
+  logger.info('connectivity rules are disabled.')
+end
 
 # WEDGE
-if $wedge == "false"
-  WEDGE = $wedge
-else
-  WEDGE = "true"
-end # WEDGE
+WEDGE = if $wedge == 'false'
+          $wedge
+        else
+          'true'
+        end
 
-logger.info("Wedge enabled  %s" % [WEDGE])
+logger.info("Wedge enabled: #{WEDGE}")
 
 # BALL
-if $ball == "false"
-  BALL = $ball
-else
-  BALL = "true"
-end # BALL
+BALL = if $ball == 'false'
+         $ball
+       else
+         'true'
+       end
 
-logger.info("Ball enabled  %s" % [BALL])
+logger.info("Ball enabled: #{BALL}")
 
 # GOLD
-if $gold == "false"
-  GOLD = $gold
-else
-  GOLD = "true"
-end # GOLD
+GOLD = if $gold == 'false'
+         $gold
+       else
+         'true'
+       end
 
-logger.info("Gold enabled  %s" % [GOLD])
+logger.info("Gold enabled: #{GOLD}")
 
-if $mim_option
-  MIM_OPTION = $mim_option
-else
-  MIM_OPTION = "B"
-end
+# MIM
+MIM_OPTION = $mim_option || 'B'
 
-logger.info("MIM Option selected %s" % [MIM_OPTION])
+logger.info("MIM Option selected: #{MIM_OPTION}")
 
 # OFFGRID
-if $offgrid == "false"
-  OFFGRID = false
-else
-  OFFGRID = true
-end # OFFGRID
+OFFGRID = $offgrid != 'false'
 
-logger.info("Offgrid enabled  %s" % [OFFGRID])
+logger.info("Offgrid enabled:  #{OFFGRID}")
 
-if $thr
-    threads($thr)
-else
-    thr ||=  Etc.nprocessors
-    threads(thr)
-end
-logger.info("Number of threads to use %s" % [$thr])
+# threads
+$thr ||= Etc.nprocessors
```

Comment:
Can we keep the original structure of the code then and keep any config refactoring for #37?

Reply:
Done, referring to #37